### ISSUE TITLE
Does undistorting an image with the colab notebook convert the camera into a pinhole?

### ISSUE THREAD

**Issue by AntonioMacaronio** [facebookresearch/projectaria_tools#65]
I am following the code in the `dataprovider_quickstart_tutorial.ipynb` colab notebook for undistorting an image, but I am curious on how this works - does this convert the image into one taken by a pinhole camera with a circular mask?

If it does convert it into an image which can be taken by a pinhole camera, is there a way to recover the intrinsic matrix of this pinhole camera?

**Comment by SeaOtocinclus** [comment]
Hello @AntonioMacaronio 
You are right about what the code is doing, you are constructing a Pinhole camera and get the undistorted image for this just defined pinhole camera.

In this code sample we used fixed parameters set by hand, but you can also directly used the one from the Fisheye camera calibration (but do care about using the right focal lenght - to get more or less pixel reprojected, according to your need)

So for example you can do, something close to the following:
```
image = self._vrs_data_provider.get_image_data_by_time_ns(
  stream_id,
  device_timestamp_ns,
  TimeDomain.DEVICE_TIME,
  TimeQueryOptions.CLOSEST,
)

focal_lengths = camera_calibration.get_focal_lengths()
image_size = camera_calibration.get_image_size()
pinhole_calib = calibration.get_linear_camera_calibration(
  image_size[0], image_size[1], focal_lengths[0]
)

# Compute the actual undistorted image
undistorted_image = distort_by_calibration(
  image[0].to_numpy_array(), pinhole_calib, camera_calibration
)
```

Using a multiplier factor on `focal_lengths[0]` will keep less or more rays of the initial camera (i.e you can use 0.8 or 1.2 and see the difference)

**Comment by AntonioMacaronio** [comment]
Gotcha, thank you so much; I need to get the intrinsic matrix for this pinhole camera though; is it safe to assume that the principal point offset of the intrinsic matrix is right in the center of the image?

in other words; the intrinsic matrix has 4 parameters f_x, f_y, c_x, c_y. can I assume the c_x =  image_width / 2, c_y = image_height / 2 ?

**Comment by SeaOtocinclus** [comment]
Yes, it is safe to assume that principal point is the center of the image when you are manipulating the undistorded image camera model (the pinhole camera model).
I will close the issue, and feel free to re-open or ask new questions if you need.
